fw/rbcx-coprocessor/src/ControlLink.cpp

```diff
@@ -89,3 +88,27 @@ bool controlLinkRx(EspMessage& incoming) {
     }
     return false;
 }
+
+void controlLinkTx(const CoprocStat& outgoing) {
+    auto encodedSize = codec.encodeWithHeader(
+        &CoprocStat_msg, &outgoing, txEncodeBuf.data(), txEncodeBuf.size());
+
+    auto sendResult
+        = xMessageBufferSend(txMessageBuf, txEncodeBuf.data(), encodedSize, 0);
```

Comment:
Dotaz ze zvědavosti. Proč zde potřebujeme kopírovat data do `txMessageBuf` a pak je v přerušení od tama vyčítat. Měli jsme s předchozí implementací nějaký problém?

Reply:
Protože tohle řeší asynchronní neblokující odesílání bez čekání na dokončení předchozího

Reply:
Messagebuffer je krásně použitelná věc, jenom to obnáší dvě kopie navíc...